spec/factories/target.rb

```diff
@@ -65,6 +45,49 @@
       end
     end
 
+    trait :with_shared_assignment do
+      transient do
+        shared_assignment { true }
+        given_role { nil }
+        given_milestone_number { nil }
+        given_evaluation_criteria { nil }
+        given_checklist { nil }
+        given_prerequisite_targets { nil }
+        given_quiz { nil }
+        with_quiz {nil}
+        with_evaluation_criterion {false}
+        with_completion_instructions { nil }
+      end
```

Comment:
I started writing it without transient. I noticed in practise that in almost all the specs targets have no attributes, but assignments have couple of attributes. So instead of just repeatedly writing `create :target, create :assignment` I thought we would combine them into a single create statement. Since a ton of specs had the `create :target` single statement it was easy to rewrite them in bulk into the new single `create :target :with_shared_assignment` statements

Reply:
@harigopal I have changed the target factory to just use role, milestone, evaluation_criteria. These are quite commonly used across all specs. In all other places I am creating the target and assignment with separate call.